src/rules/switchDefaultRule.ts

```diff
@@ -15,41 +15,216 @@
  * limitations under the License.
  */
 
-import { isDefaultClause } from "tsutils";
+import { inspect } from "util";
+
+import {
+    isDefaultClause,
+    isLiteralType,
+    isSwitchStatement,
+    isTypeFlagSet,
+    unionTypeParts,
+} from "tsutils";
+
 import * as ts from "typescript";
 
+import { showWarningOnce } from "../error";
 import * as Lint from "../index";
+import { hasOwnProperty } from "../utils";
 import { codeExamples } from "./code-examples/switchDefault.examples";
 
-export class Rule extends Lint.Rules.AbstractRule {
+const OPTION_ALLOW_EXHAUSTIVE = "allow-exhaustive";
+
+interface Options {
+    allowExhaustive: boolean;
+}
+
+export class Rule extends Lint.Rules.OptionallyTypedRule {
     /* tslint:disable:object-literal-sort-keys */
     public static metadata: Lint.IRuleMetadata = {
         ruleName: "switch-default",
         description: "Require a `default` case in all `switch` statements.",
-        optionsDescription: "Not configurable.",
-        options: null,
-        optionExamples: [true],
+        optionsDescription: Lint.Utils.dedent`
+            One argument may be optionally provided:
```

Comment:
TSLint is moving towards having options provided in objects, which makes it easier for rules to move towards more advanced configurations. See `trailingCommaRule.ts` for an example.